src/System.Net.Sockets.Legacy/tests/FunctionalTests/SendReceive.cs

```diff
@@ -174,6 +175,7 @@ private static void SendToRecvFromAPM_Datagram_UDP(IPAddress leftAddress, IPAddr
                     if (sentDatagrams == DatagramsToSend)
                     {
                         right.Dispose();
+                        senderFinished.SetResult(true);
                         return;
                     }
```

Comment:
Should we wrap the body of this if with a try/catch that stores any exceptions (including from an assert) into the senderFinished tcs?  Then such errors would come out of the Wait call in the main body of the test.